src/core/shape_inference/include/generate_proposals_shape_inference.hpp

```diff
@@ -22,106 +22,120 @@ std::vector<TRShape> shape_infer(const GenerateProposals* op, const std::vector<
     }
     const auto deltas_shape_rank = deltas_shape.rank();
     const auto scores_shape_rank = scores_shape.rank();
 
-    NODE_VALIDATION_CHECK(op,
-                          deltas_shape_rank.compatible(4),
-                          "The 'input_deltas' input is expected to be a 4D. Got: ",
-                          deltas_shape);
-    NODE_VALIDATION_CHECK(op,
-                          scores_shape_rank.compatible(4),
-                          "The 'input_scores' input is expected to be a 4D. Got: ",
-                          scores_shape);
+    NODE_SHAPE_INFER_CHECK(op,
+                           input_shapes,
+                           deltas_shape_rank.compatible(4),
+                           "The 'input_deltas' input is expected to be a 4D. Got: ",
+                           deltas_shape);
+    NODE_SHAPE_INFER_CHECK(op,
+                           input_shapes,
+                           scores_shape_rank.compatible(4),
+                           "The 'input_scores' input is expected to be a 4D. Got: ",
+                           scores_shape);
     if (deltas_shape_rank.is_static())
         Dimension::merge(num_batches, deltas_shape[0], num_batches);
     if (scores_shape_rank.is_static())
         Dimension::merge(num_batches, scores_shape[0], num_batches);
 
     if (deltas_shape_rank.is_static() && scores_shape_rank.is_static()) {
-        NODE_VALIDATION_CHECK(op,
-                              deltas_shape[0].compatible(scores_shape[0]),
-                              "Batch for inputs 'input_deltas' and 'input_scores' should be "
-                              "equal. Got: ",
-                              deltas_shape[0],
-                              scores_shape[0]);
-
-        NODE_VALIDATION_CHECK(op,
-                              deltas_shape[1].compatible(scores_shape[1] * 4),
-                              "Anchor number for inputs 'input_deltas' and 'input_scores' should be "
-                              "equal. Got: ",
-                              deltas_shape[1] / 4,
-                              scores_shape[1]);
-
-        NODE_VALIDATION_CHECK(op,
-                              deltas_shape[2].compatible(scores_shape[2]),
-                              "Heights for inputs 'input_deltas' and 'input_scores' should be "
-                              "equal. Got: ",
-                              deltas_shape[2],
-                              scores_shape[2]);
-
-        NODE_VALIDATION_CHECK(op,
-                              deltas_shape[3].compatible(scores_shape[3]),
-                              "Width for inputs 'input_deltas' and 'input_scores' should be "
-                              "equal. Got: ",
-                              deltas_shape[3],
-                              scores_shape[3]);
+        NODE_SHAPE_INFER_CHECK(op,
+                               input_shapes,
+                               deltas_shape[0].compatible(scores_shape[0]),
```

Comment:
I think, the merges above could be used here instead compatible check.

Reply:
Yes, seems like Dimension::merge and check can be done at once, but I would like to avoid unnecessary changes in this PR.
Let's merge current version and apply other suggestions separately.